macros/models/_sector/nft/enrich_nft_trades.sql

```diff
@@ -0,0 +1,108 @@
+    {% if is_incremental() %}
+    AND {{incremental_predicate('minute')}}
+    {% endif %}
+    UNION ALL
+    SELECT
+        {{ var("ETH_ERC20_ADDRESS") }} as contract_address
+        ,'ethereum' as blockchain
+        ,18 as decimals
+        ,minute
+        ,price
+        ,'ETH' as symbol
+    FROM {{ ref('prices_usd_forward_fill') }}
+    WHERE blockchain is null AND symbol = 'ETH'
+    {% if is_incremental() %}
+    AND {{incremental_predicate('minute')}}
+    {% endif %}
+),
+enrichments as (
+SELECT
+    base.blockchain,
+    base.project,
+    base.project_version,
+    base.block_date,
+    base.block_month,
+    base.block_time,
+    base.block_number,
+    base.tx_hash,
+    base.sub_tx_trade_id,
+    base.project_contract_address,
+    base.trade_category,
+    base.trade_type,
+    case when base.buyer = coalesce(agg1.contract_address,agg2.contract_address) then base.tx_from else base.buyer end as buyer,
+    case when base.seller = coalesce(agg1.contract_address,agg2.contract_address) then base.tx_from else base.seller end as seller,
+    base.nft_contract_address,
+    base.nft_token_id,
+    base.nft_amount,
+    base.price_raw,
+    base.currency_contract,
+    base.platform_fee_amount_raw,
+    base.royalty_fee_amount_raw,
+    base.platform_fee_address,
+    base.royalty_fee_address,
+    base.tx_from as tx_from,
+    base.tx_to as tx_to,
+    nft.name as nft_collection,
+    nft.standard as nft_standard,
+    coalesce(erc20.symbol,p.symbol) as currency_symbol,
+    base.price_raw/pow(10,coalesce(erc20.decimals,p.decimals,18)) as price,
+    base.platform_fee_amount_raw/pow(10,coalesce(erc20.decimals,p.decimals,18)) as platform_fee_amount,
+    base.royalty_fee_amount_raw/pow(10,coalesce(erc20.decimals,p.decimals,18)) as royalty_fee_amount,
+    base.price_raw/pow(10,coalesce(erc20.decimals,p.decimals,18))*p.price as price_usd,
+    base.platform_fee_amount_raw/pow(10,coalesce(erc20.decimals,p.decimals,18))*p.price as platform_fee_amount_usd,
+    base.royalty_fee_amount_raw/pow(10,coalesce(erc20.decimals,p.decimals,18))*p.price as royalty_fee_amount_usd,
+    case when base.price_raw > uint256 '0' then cast(100*base.platform_fee_amount_raw/base.price_raw as double) else double '0' end as platform_fee_percentage,
+    case when base.price_raw > uint256 '0' then cast(100*base.royalty_fee_amount_raw/base.price_raw as double) else double '0' end as royalty_fee_percentage,
+    coalesce(agg1.contract_address,agg2.contract_address) as aggregator_address,
+    {% if aggregator_markers != null %}
+    CASE WHEN coalesce(agg_mark.aggregator_name, agg1.name, agg2.name)='Gem' AND base.block_number >= 16971894 THEN 'OpenSea Pro' -- 16971894 is the first block of 2023-04-04 which is when Gem rebranded to OpenSea Pro
+        ELSE coalesce(agg_mark.aggregator_name, agg1.name, agg2.name)
+        END as aggregator_name
+    {% else %}
+    coalesce(agg1.name,agg2.name) as aggregator_name
+    {% endif %}
+FROM {{base_trades}} base
```

Comment:
i was wondering why the final step took so long in incremental run -- looks like you're missing incremental logic / filter on this base table?

Reply:
Thank you, I was looking for the reason of those ~15min incremental steps, but wasn't able to find it.